### ISSUE TITLE
Configure a WASM filter at the Route level

### ISSUE THREAD

**Issue by djannot** [kgateway-dev/kgateway#5426]
**Is your feature request related to a problem? Please describe.**
Currently, a WASM filter is applied at the Gateway level and is used for all the requests going through this gateway.

**Describe the solution you'd like**
Being able to apply a WASM filter at the Route level

**Additional context**
This feature request also depends on the ability for Envoy to do that and there's an issue related to it:
https://github.com/envoyproxy/envoy/issues/16651
It doesn't seem to be really moving, so perhaps we should contribute on that side

**Comment by chrisgaun** [comment]
@yuval-k has good idea of what is missing here.

**Comment by yuval-k** [comment]
would this help?
https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/composite_filter.html

**Comment by jrauschenbusch** [comment]
@yuval-k Yes, i think this can help as long as there is no ability of using `per_filter_config`/`typed_per_filter_config` for wasm filters.

> Filesystem config update rejected: Error adding/updating listener(s) listener_https: The filter envoy.filters.http.wasm doesn't support virtual host-specific configurations

But i think it would be slightly simpler for most scenarios to have support for `per_filter_config`. But it seems all related tickets were closed for some reason or there is no real progress:
- https://github.com/envoyproxy/envoy/issues/16651
- https://github.com/envoyproxy/envoy/issues/15129
- https://github.com/envoyproxy/envoy/issues/11690

For Lua filter there is already support: https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/lua_filter#per-route-configuration

**Comment by yuval-k** [comment]
That's correct, this is a workaround - the config will be in the filter, and the filter itself will be enabled/disabled based on route properties

**Comment by jaymiracola** [comment]
We have another customer interested in this to decode urls into strings.
